lib/intercom/search_collection_proxy.rb

```diff
@@ -0,0 +1,82 @@
+require "intercom/utils"
+
+module Intercom
+  class SearchCollectionProxy
+
+    attr_reader :resource_name, :resource_class
+
+    def initialize(resource_name, search_details: {}, client:)
+      @resource_name = resource_name
+      @resource_class = Utils.constantize_resource_name(resource_name)
+      @search_url = search_details[:url]
+      @search_params = search_details[:params]
+      @client = client
+    end
+
+    def each(&block)
+      loop do
+        response_hash = @client.post(@search_url, payload)
```

Comment:
Is it standard to not set the client to be an `attr_reader`?

Reply:
I've based this on other collection proxy, which do not set the client here. I guess that's fine - as long as we can run the search query?

Reply:
Yep I reckon so! Was just wondering the reasoning behind it 😄